src/Sylius/Bundle/ApiBundle/DataProvider/CurrencyCollectionDataProvider.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\ApiBundle\DataProvider;
+
+use ApiPlatform\Core\DataProvider\CollectionDataProviderInterface;
+use ApiPlatform\Core\DataProvider\RestrictedDataProviderInterface;
+use Sylius\Bundle\ApiBundle\Context\UserContextInterface;
+use Sylius\Bundle\ApiBundle\Serializer\ContextKeys;
+use Sylius\Component\Core\Model\AdminUserInterface;
+use Sylius\Component\Core\Model\ChannelInterface;
+use Sylius\Component\Currency\Model\CurrencyInterface;
+use Sylius\Component\Resource\Repository\RepositoryInterface;
+use Webmozart\Assert\Assert;
+
+/** @experimental */
+final class CurrencyCollectionDataProvider implements CollectionDataProviderInterface, RestrictedDataProviderInterface
+{
+    private UserContextInterface $userContext;
+    private RepositoryInterface $currencyRepository;
+
+    public function __construct(
+        RepositoryInterface $currencyRepository,
+        UserContextInterface $userContext
+    ) {
+        $this->userContext = $userContext;
+        $this->currencyRepository = $currencyRepository;
+    }
+
+    public function supports(string $resourceClass, string $operationName = null, array $context = []): bool
```

Comment:
when a method have 3 or more arguments, switch to multiline

Reply:
we are applying this rule only when the method does not fit regular github screen (around 80 characters)